src/main/resources/assets/computercraft/lua/rom/programs/list.lua

```diff
@@ -2,8 +2,27 @@
 local tArgs = { ... }
 
 -- Get all the files in the directory
-local sDir = shell.dir()
-if tArgs[1] ~= nil then
+local sDir
+if #tArgs == 0 then
+    sDir = shell.dir()
+elseif string.sub(tArgs[1],1,1) == "-" then
```

Comment:
This argument parsing code doesn't handle several cases you might expect it to do, such as `ls -l -a foo`, or `ls foo -a`.

I wonder if it's worth adding something like cloud-catcher's [arg-parse] to CC, which would mean we can have something which is a little more general and well behaved.

[arg-parse]: https://github.com/SquidDev-CC/cloud-catcher/blob/master/src/host/argparse.lua

Reply:
Probably worth thinking about how the shell auto-completion should handle arguments too. Currently `ls -l f` won't auto-complete into `ls -l foo`.